api.planx.uk/send/bops.js

```diff
@@ -1,27 +1,37 @@
-import {
-  responseInterceptor,
-  fixRequestBody,
-} from "http-proxy-middleware";
-import { adminGraphQLClient } from "../hasura";
+import { responseInterceptor } from "http-proxy-middleware";
+import { adminGraphQLClient as client } from "../hasura";
 import { useProxy } from "../server";
 import { markSessionAsSubmitted } from "../saveAndReturn/utils";
 
-const client = adminGraphQLClient;
+const sendToBOPS = async (req, res, next) => {
+  // `/bops/:localAuthority` is only called via Hasura's scheduled event webhook now, so body is wrapped in a "payload" key
+  const { payload } = req.body;
+  if (!payload) {
+    next({
+      status: 400,
+      message: `Missing application payload data to send to BOPS`,
+    });
+  }
+
+  // confirm that this session has not already been successfully submitted before proceeding
+  const submittedApp = await checkBOPSAuditTable(payload?.planx_debug_data?.session_id);
+  if (submittedApp?.message === "Application created") {
```

Comment:
this check ensures that we will still proceed & "retry" for failed submissions. 

as an example, see BOPS prod case from last week that we have multiple audit entries for, but all prior ones have response `{ "message": "Validation failed: File The selected file must be a PDF, JPG or PNG" }`

Reply:
This will certainly work, but does leave us vulnerable to a BOPS change here. This seems unlikely, but if there's a way we could also check a HTTP response code or similar we could have a bit more confidence here.

Reply:
I think we could capture this from BOPS and keep it in the table, as these changes are really aimed at future duplicate applications which happen in close succession as opposed to checking an historic entry vs a recent one.

Reply:
Right, we don't store a status code currently, just this message response and some other not-so-useful response headers.

In the future, I was thinking the interesting improvement here could be moving away from querying the audit tables altogether and actually just checking if Hasura already has an event queued up for this session id & destination. To do that, I think we'd want to create a tracked view that pulls event info from the pg metadata tables that we could then query via GraphQL. I think it'd have the extra benefit of possibly catching more cases, as the event is created far before the audit entry is inserted.